pkg/utils/resource/update/update.go

```diff
@@ -0,0 +1,72 @@
+package update
+
+import (
+	"context"
+	"reflect"
+
+	cco "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/aws-load-balancer-operator/pkg/utils"
+)
+
+// CredentialsRequest updates the given CredentialsRequest to the desired state if needed.
+// Returns true if the CredentialsRequest was updated, false otherwise.
+func CredentialsRequest(ctx context.Context, client client.Client, current *cco.CredentialsRequest, desired *cco.CredentialsRequest) (bool, error) {
```

Comment:
Along with my style comment above, more specifically I feel like this could be better classified as an `ensure` operation vs. an `update`. Ensure implying it'll ensure it will get the object into the right state, but it may not need to update it, whereas update implies its going to always update it (which sometimes it doesn't here).

Reply:
Looking at the ensuring functions in this repo ([example](https://github.com/openshift/aws-load-balancer-operator/blob/main/pkg/controllers/awsloadbalancercontroller/credentials_request.go#L39)): it has to do `get-create/update-get`. At first I thought of moving the `ensureCredentialsRequest` function into this package but that would need too many adaptations to make it work in both: main and controller. So I took the function which needs the least changes for the existing code but does its job for the new code.

Reply:
Yea I see now. Seems reasonable. There's some refactoring you could probably do, but would require an overhaul. I think this is acceptable for now.

I do find it bit funky that the function name `CredentialsRequest` doesn't completely describe what the function does, but it needs the package name `update` to aptly describe it. If you are local, like in `credentials_request_test.go`, then you won't call the function with the package name (`update.CredentialsRequest` vs. `CredentialsRequest`). Is this pattern used in other places?

Reply:
Okay I see, it's debatable. On the side of brevity, you are right, I agree with https://go.dev/doc/effective_go#package-names. But in the examples I see, the package name was the "noun" and the function was the "verb"; whereas, this is backwards (i.e. package `bufio` and function `reader` vs. package `update` and function `CredentialsRequest`).

I don't think this is important enough to hold up the PR. I'll LGTM and approve and you can proceed how you wish.